framework/db/poutput_manager.py

```diff
@@ -123,21 +123,34 @@ def GetAll(self, filter_data=None, target_id=None):
         return self.DeriveOutputDicts(results, target_id=target_id)
 
     @target_required
+    def GetAllPluginNames(self, filter_data=None, target_id=None):
```

Comment:
Ok +1. Write comments (inline and header) so that it is easier to review and modify in the future.